### ISSUE TITLE
Add end-to-end tests for HTTP/2

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#2170]
This would probably use netty HTTP/2 client for the time being to end-to-end test the HTTP/2 server (configuration).

Also, make sure to find problems like the `parallelism=0` problem discussed in #2168.

**Comment by raboof** [comment]
We also have the 'h2spec' tests that are fairly end-to-end, right?

**Comment by jrudolph** [comment]
> We also have the 'h2spec' tests that are fairly end-to-end, right?

Right, but they mostly test somewhat rare edge cases that we don't support completely so we mostly ignore the output...

**Comment by jrudolph** [comment]
> We also have the 'h2spec' tests that are fairly end-to-end, right?

Also h2spec only tests the server but not the client.
